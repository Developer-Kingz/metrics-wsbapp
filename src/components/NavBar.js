import React from 'react';
import PropTypes from 'prop-types';

const NavBar = ({ search, onSearch }) => (
  <div className="navbar">
    <div className="nav-inner">
      <img src="https://cdn.britannica.com/94/75394-050-C3F58654/World-map-Continents-Oceans-Seas.jpg" alt="" className="nav-img" />
      <h4>Search Countries</h4>
    </div>
    <input
      type="text"
      value={search}
      onChange={onSearch}
      placeholder="Search Country"
      className="search-country"
    />
  </div>
);

export default NavBar;
NavBar.propTypes = {
  search: PropTypes.string.isRequired,
  onSearch: PropTypes.func.isRequired,
};
