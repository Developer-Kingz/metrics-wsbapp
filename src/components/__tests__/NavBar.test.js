import React from 'react';
import { render } from '@testing-library/react';
import NavBar from '../NavBar';

test('NavBar matches snapshot', () => {
  const search = '';
  const onSearch = jest.fn();

  const { asFragment } = render(<NavBar search={search} onSearch={onSearch} />);

  expect(asFragment()).toMatchSnapshot();
});
