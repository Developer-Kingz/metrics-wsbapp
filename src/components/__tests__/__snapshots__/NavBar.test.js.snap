// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="navbar"
  >
    <div
      class="nav-inner"
    >
      <img
        alt=""
        class="nav-img"
        src="https://cdn.britannica.com/94/75394-050-C3F58654/World-map-Continents-Oceans-Seas.jpg"
      />
      <h4>
        Search Countries
      </h4>
    </div>
    <input
      class="search-country"
      placeholder="Search Country"
      type="text"
      value=""
    />
  </div>
</DocumentFragment>
`;
